libraries/AP_HAL_SITL/SITL_State.cpp

```diff
@@ -207,6 +207,7 @@ void SITL_State::wait_clock(uint64_t wait_time_usec)
             _fdm_input_step();
         } else {
             usleep(1000);
+            wait_time_usec = MAX(0U, wait_time_usec-1000);
```

Comment:
I believe this is incorrect, as this marches the final absolute time forward as time marches, and effectively means we return in half the requested time. This is only ever called from `delay_microseconds` which also does a while loop, which I think masked the error this would introduce. That being said this should just be inlined up instead of nesting the two loops, as I can't see a reason for this.